src/apps/external_executor/protobuf/index.proto

```diff
@@ -0,0 +1,74 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+
+package externalexecutor.protobuf;
+
+option optimize_for = LITE_RUNTIME;
+
+// Service exposed by a CCF node for Executors/Indexers to interact with the
+// Index stored on the CCF node. Communication with this service must be over
+// TLS, authorising as a client cert that has previously been accepted by the
+// ExecutorRegistration service.
+service Index
+{
+  // Installs index and registers with the given strategy name and table name.
+  // Streams Key-Value pairs of every transaction in the given table name.
+  // Stores indexed data in an unordered map datastructure or a prefix tree
+  // depending on the type of enum passed
+  rpc InstallAndSubscribe(IndexInstall) returns (stream IndexWork) {}
+
+  // Stores Indexer's data. Can only be invoked by the Indexer that installed
+  // the strategy
+  rpc StoreIndexedData(IndexKeyValue) returns (google.protobuf.Empty) {}
```

Comment:
The C++ implementation uses `IndexPayload` for the input type here, rather than `IndexKeyValue`. The former includes the `strategy_name`, so allows an executor to implement multiple strategies. Without that, I think we're unnecessarily restricting to a single index-per-executor, and not matching that in the implementation.

Reply:
`IndexPayload` is the correct and intended type (and so is the cpp implementation). Will correct this in the proto file.